lib/components/viewers/route-details.tsx

```diff
@@ -87,33 +93,44 @@ class RouteDetails extends Component<Props> {
   }
 
   render() {
-    const { intl, operator, patternId, route, setHoveredStop } = this.props
+    const {
+      intl,
+      operator,
+      patternId,
+      route,
+      setHoveredStop,
+      sortPatternsByVehicleCount
+    } = this.props
     const { agency, patterns = {}, shortName, url } = route
     const pattern = patterns[patternId]
 
     const moreDetailsURL = url || route?.agency?.url
 
     const routeColor = getRouteColorBasedOnSettings(operator, route)
 
-    const headsigns = extractMainHeadsigns(
+    const unsortedHeadsigns = extractMainHeadsigns(
       patterns,
       shortName,
       this._editHeadsign
-    ).sort((a, b) => {
-      // sort by number of vehicles on that pattern
-      const aVehicleCount =
-        route.vehicles?.filter((vehicle) => vehicle.patternId === a.id)
-          .length || 0
-      const bVehicleCount =
-        route.vehicles?.filter((vehicle) => vehicle.patternId === b.id)
-          .length || 0
-
-      // if both have the same count, sort by pattern geometry length
-      if (aVehicleCount === bVehicleCount) {
-        return b.geometryLength - a.geometryLength
-      }
-      return bVehicleCount - aVehicleCount
-    })
+    )
+
+    const headsigns = sortPatternsByVehicleCount
+      ? unsortedHeadsigns.sort((a, b) => {
+          // sort by number of vehicles on that pattern
```

Comment:
We can avoid adding the new variable and the ternary if you instead add an `if(!sortPatternsByVehicleCount) return 0` at the top of the original sort method